src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -126,15 +127,17 @@ public function decode($data, $format, array $context = array())
                 $nbHeaders = $nbCols;
 
                 foreach ($cols as $col) {
-                    $headers[] = explode($keySeparator, $col);
+                    $headerChain = explode($keySeparator, $col);
+                    $headers[] = $headerChain;
+                    $headersDepth[] = count($headerChain);
```

Comment:
The `$headersDepth` variable name is a bit confusing because of the `Depth` word. Could we rename it to `$headerCount` or `$numOfHeaders`, etc.?